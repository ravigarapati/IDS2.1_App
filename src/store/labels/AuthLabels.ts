export const CURRENT_USER = 'CURRENT_USER';
export const LOGOUT = 'LOGOUT';
export const CURRENT_STEP = 'CURRENT_STEP';
export const UPDATE_CURRENT_USER = 'UPDATE_CURRENT_USER';
export const SHOW_LOADING = 'SHOW_LOADING';
export const TERMS_AND_CONDITIONS = 'TERMS_AND_CONDITIONS';
export const LATEST_VERSION = 'LATEST_VERSION';
export const SET_STEP_TITLES = 'SET_STEP_TITLES';
export const SET_USER_ROLE = 'SET_USER_ROLE';
export const SET_SHOW_ROLE = 'SET_SHOW_ROLE';
export const SET_USER_NAME = 'SET_USER_NAME';
export const SET_USER_ADDRESS = 'SET_USER_ADDRESS';
export const RESET_USER = 'RESET_USER';
