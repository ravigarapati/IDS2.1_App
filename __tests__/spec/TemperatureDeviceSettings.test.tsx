import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import configureMockStore from 'redux-mock-store';
import {Provider} from 'react-redux';
import TemperatureDeviceSettings from '../../src/pages/TemperatureDeviceSettings';
import {render, cleanup, fireEvent} from 'react-native-testing-library';
import HomeOwnerReducer from './../../src/store/reducers/HomeOwnerReducer';
import * as homeOwnerActions from './../../src/store/actions/HomeOwnerActions';
import {createStore, combineReducers, applyMiddleware} from 'redux';
import thunk from 'redux-thunk';
import {UPDATE_TEMPERATURE_SETTINGS} from '../../src/store/labels/HomeOwnerLabels';

jest.useFakeTimers();

const mockStore = configureMockStore();
const store = mockStore({
  homeOwner: {
    energyData: {},
    deviceList: [],
    selectedDevice: {},
    contractorMonitoringStatus: false,
    fanMode: '',
    fanOnFor: '',
    fanOffFor: '',
    fanIsScheduled: '',
    fanScheduledStart: '',
    fanScheduledEnd: '',
    selectedSchedule: '0',
    selectedScheduleName: '',
    scheduleInfo: {},
    prevSelectedDevice: {},
    deviceStatus: {},
    deviceInformation: {},
    deviceDetails: {},
    heatPumpInfo: {},
    reloadHeatPumpInfo: false,
    notificationList: [],
    lastKey: {},
    notificationsCount: 0,
    lastUpdated: null,
    isTermsConditionsAccepted: null,
    hoUserAnalytics: null,
    selectedUnitName: {},
    isThermostatSelected: true,
    isUserFirstLogin: true,
    actualWeatherOnFahrenheit: true,
    hapticVibration: true,
    _24HrsFormatSelected: false,
    location: 'Alabaster',
    updatedSelectedDevice: false,
    weatherInfo: {},
    weatherInfoLocation: {},
    locationSuggestions: [],
    locationInformation: {},
    idsSelectedDeviceAccess: '',
    utilyEnergySaving: '',
    idsSelectedDevice: '',
    idsSelectedOdu: '',
    idsSelectedDeviceType: '',
    tempLocation: '',
    newDeviceInfo: {},
    previousBcc: '',
    deviceList2: [
      /*{
        id: 2,
        //macId: 'e8fdf8a4773c',
        macId: '34eae7c351e6',
        deviceName: 'Bedroom Thermostat',
        deviceType: 'BCC100',
        isThermostat: true,
        isOn: true,
        mode: 0,
        setPoint: 74,
        heatingSetpoint: 74,
        coolingSetpoint: 79,
        current: 76,
        isAccesories: true,
        accesories: [
          {
            id: 0,
            name: 'Dehumidifier',
            isEnable: 0,
            isOn: false,
          },
          {
            id: 1,
            name: 'Humidifier',
            isEnable: 1,
            isOn: false,
          },
        ],
        isMonitoring: null,
        isOnSchedule: true,
        stage: 3,
        isConnected: true,
        acceleratedHeating: true,
        schedules: [],
      },*/
    ],
  },
  auth: {},
  contractor: {
    selectedUnit: {
      odu: {
        modelNumber: 'testing',
      },
      systemStatus: 'testing',
      gateway: {gatewayId: 'testing'},
    },
  },
});

const rootReducer = combineReducers({
  homeOwner: HomeOwnerReducer,
});

const realStore = createStore(
  rootReducer,
  {
    homeOwner: {
      energyData: {},
      deviceList: [],
      selectedDevice: {
        autoOn: false,
        alertMessage: false,
        alarmLow: '70',
        alarmHigh: '80',
        lockDevice: false,
        d_hour: '1',
        isFahrenheit: true,
        deadband: 3,
        schedules: [
          {
            default_home: '1',
            limit: '71-81',
            mode: '1',
            model_id: '1',
            name: 'Home',
            state: '0',
          },
          {
            default_home: '1',
            limit: '71-81',
            mode: '1',
            model_id: '2',
            name: 'Vacation',
            state: '0',
          },
          {
            default_home: 0,
            limit: '71-82',
            mode: '3',
            model_id: '3',
            name: 'summer ',
            state: '1',
          },
          {
            default_home: 0,
            limit: '71-82',
            mode: '3',
            model_id: '4',
            name: 'summer cam',
            state: '0',
          },
        ],
      },
      contractorMonitoringStatus: false,
      fanMode: '',
      fanOnFor: '',
      fanOffFor: '',
      fanIsScheduled: '',
      fanScheduledStart: '',
      fanScheduledEnd: '',
      selectedSchedule: '1',
      selectedScheduleName: '',
      scheduleInfo: {
        items1: [{c: '0', h: '0', t: '78.0-70.0'}],
        items2: [
          {c: '0', h: '0', t: '78.0-73.0'},
          {c: '0', h: '0', t: '99.0-90.0'},
        ],
        items3: [{c: '0', h: '1185', t: '52.0-48.0'}],
        items4: [{c: '0', h: '0', t: '78.0-70.0'}],
        items5: [{c: '0', h: '0', t: '78.0-70.0'}],
        items6: [{c: '0', h: '0', t: '78.0-70.0'}],
        items7: [{c: '0', h: '0', t: '78.0-70.0'}],
      },
      prevSelectedDevice: {},
      deviceStatus: {},
      deviceInformation: {},
      deviceDetails: {},
      heatPumpInfo: {},
      reloadHeatPumpInfo: false,
      notificationList: [],
      lastKey: {},
      notificationsCount: 0,
      lastUpdated: null,
      isTermsConditionsAccepted: null,
      hoUserAnalytics: null,
      selectedUnitName: {},
      isThermostatSelected: true,
      isUserFirstLogin: true,
      actualWeatherOnFahrenheit: true,
      hapticVibration: true,
      _24HrsFormatSelected: false,
      location: 'Alabaster',
      updatedSelectedDevice: false,
      weatherInfo: {},
      weatherInfoLocation: {},
      locationSuggestions: [
        {
          PlaceId:
            'AQABAFMAKZmJqReTuJvQ0SYSvNXBFPnTNre8NpZN_vO5ppVBiIVKMWyFo7U_23WZIEB3jBaCX9aLHNHy6gHy6QyfVv1o8QfoXctScRaEWzssnUy2A7vSsswLy5wkja6Jfp6hQXlv4QvGMRLRNcLPAogsltpxpywjew',
          Text: 'New York, NY, United States',
        },
      ],
      locationInformation: {},
      idsSelectedDeviceAccess: '',
      utilyEnergySaving: '',
      idsSelectedDevice: '',
      idsSelectedOdu: '',
      idsSelectedDeviceType: '',
      tempLocation: '',
      newDeviceInfo: {},
      previousBcc: '',
      deviceList2: [
        /*{
        id: 2,
        //macId: 'e8fdf8a4773c',
        macId: '34eae7c351e6',
        deviceName: 'Bedroom Thermostat',
        deviceType: 'BCC100',
        isThermostat: true,
        isOn: true,
        mode: 0,
        setPoint: 74,
        heatingSetpoint: 74,
        coolingSetpoint: 79,
        current: 76,
        isAccesories: true,
        accesories: [
          {
            id: 0,
            name: 'Dehumidifier',
            isEnable: 0,
            isOn: false,
          },
          {
            id: 1,
            name: 'Humidifier',
            isEnable: 1,
            isOn: false,
          },
        ],
        isMonitoring: null,
        isOnSchedule: true,
        stage: 3,
        isConnected: true,
        acceleratedHeating: true,
        schedules: [],
      },*/
      ],
    },
  },
  applyMiddleware(thunk),
);

describe('BCC Temperature device screen', () => {
  test('Component to be defined', () => {
    const navState = {params: {device: 'BCC50'}};
    const navigation = {
      getParam: (key, val) => navState?.params[key] ?? val,
    };
    const tree = renderer
      .create(
        <Provider store={store}>
          <TemperatureDeviceSettings navigation={navigation} />
        </Provider>,
      )
      .toJSON();

    expect(tree).toBeDefined();
  });
  test('Component to be rendered', () => {
    const navState = {params: {device: 'BCC50'}};
    const navigation = {
      getParam: (key, val) => navState?.params[key] ?? val,
    };
    const tree = renderer
      .create(
        <Provider store={store}>
          <TemperatureDeviceSettings navigation={navigation} />
        </Provider>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
  test('Change Temperature', () => {
    const navState = {
      params: {
        createStatusInterval: () => {},
        setAuxHold: () => {},
        setUpdateInfo: () => {},
      },
    };
    const navigation = {
      getParam: (key, val) => navState?.params[key] ?? val,
      state: navState,
    };
    const rendered = render(
      <Provider store={store}>
        <TemperatureDeviceSettings navigation={navigation} />
      </Provider>,
    );
    //let fButton = rendered.getByTestId('button1');
    //fireEvent(fButton, 'press');
  });

  test('Enable auto option', () => {
    const navState = {
      params: {
        createStatusInterval: () => {},
        setAuxHold: () => {},
        setUpdateInfo: () => {},
      },
    };
    const navigation = {
      getParam: (key, val) => navState?.params[key] ?? val,
      state: navState,
    };
    const rendered = render(
      <Provider store={realStore}>
        <TemperatureDeviceSettings navigation={navigation} />
      </Provider>,
    );
    let updateTemp = jest
      .spyOn(homeOwnerActions, 'updateTemperatureSetting')
      .mockImplementationOnce(_ =>
        realStore.dispatch({
          type: UPDATE_TEMPERATURE_SETTINGS,
          data: {
            low: '70',
            high: '89',
            t_limit: '0',
            t_auto: '1',
          },
        }),
      );
    updateTemp();
    let fButton = rendered.getByTestId('autoOption');
    fireEvent(fButton, 'press');

    expect(realStore.getState().homeOwner.selectedDevice.autoOn).toBeTruthy();
  });

  test('Enable alert option', () => {
    const navState = {
      params: {
        createStatusInterval: () => {},
        setAuxHold: () => {},
        setUpdateInfo: () => {},
      },
    };
    const navigation = {
      getParam: (key, val) => navState?.params[key] ?? val,
      state: navState,
    };
    const rendered = render(
      <Provider store={realStore}>
        <TemperatureDeviceSettings navigation={navigation} />
      </Provider>,
    );
    let updateTemp = jest
      .spyOn(homeOwnerActions, 'updateTemperatureSetting')
      .mockImplementationOnce(_ =>
        realStore.dispatch({
          type: UPDATE_TEMPERATURE_SETTINGS,
          data: {
            low: '70',
            high: '89',
            t_limit: '1',
            t_auto: '0',
          },
        }),
      );
    updateTemp();
    let fButton = rendered.getByTestId('alertOption');
    fireEvent(fButton, 'press');

    expect(
      realStore.getState().homeOwner.selectedDevice.alertMessage,
    ).toBeTruthy();
  });

  test('Change low temp', () => {
    const navState = {
      params: {
        createStatusInterval: () => {},
        setAuxHold: () => {},
        setUpdateInfo: () => {},
      },
    };
    const navigation = {
      getParam: (key, val) => navState?.params[key] ?? val,
      state: navState,
    };
    const rendered = render(
      <Provider store={realStore}>
        <TemperatureDeviceSettings navigation={navigation} />
      </Provider>,
    );
    let updateTemp = jest
      .spyOn(homeOwnerActions, 'updateTemperatureSetting')
      .mockImplementationOnce(_ =>
        realStore.dispatch({
          type: UPDATE_TEMPERATURE_SETTINGS,
          data: {
            low: '45',
            high: '89',
            t_limit: '1',
            t_auto: '0',
          },
        }),
      );
    updateTemp();
    let fButton = rendered.getByTestId('lowTemp');
    fireEvent(fButton, 'confirm');
    expect(realStore.getState().homeOwner.selectedDevice.alarmLow).toBe('45');
  });

  test('Change high temp', () => {
    const navState = {
      params: {
        createStatusInterval: () => {},
        setAuxHold: () => {},
        setUpdateInfo: () => {},
      },
    };
    const navigation = {
      getParam: (key, val) => navState?.params[key] ?? val,
      state: navState,
    };
    const rendered = render(
      <Provider store={realStore}>
        <TemperatureDeviceSettings navigation={navigation} />
      </Provider>,
    );
    let updateTemp = jest
      .spyOn(homeOwnerActions, 'updateTemperatureSetting')
      .mockImplementationOnce(_ =>
        realStore.dispatch({
          type: UPDATE_TEMPERATURE_SETTINGS,
          data: {
            low: '45',
            high: '60',
            t_limit: '1',
            t_auto: '0',
          },
        }),
      );
    updateTemp();
    let fButton = rendered.getByTestId('highTemp');
    fireEvent(fButton, 'confirm');
    expect(realStore.getState().homeOwner.selectedDevice.alarmHigh).toBe('60');
  });
});
