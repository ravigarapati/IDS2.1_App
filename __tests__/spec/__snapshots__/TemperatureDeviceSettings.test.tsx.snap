// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BCC Temperature device screen Component to be rendered 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#ffff",
        "flexDirection": "row",
        "paddingBottom": 7,
        "paddingTop": 7,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#ffff",
          "flex": 0.15,
          "flexDirection": "column",
          "justifyContent": "center",
        }
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "flex": 1,
            "justifyContent": "center",
            "opacity": 1,
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": undefined,
                "fontSize": 24,
              },
              Object {
                "marginHorizontal": 10,
              },
              Object {
                "fontFamily": "Bosch-Icon",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "center",
          "marginLeft": 24,
        }
      }
    >
      <Text
        accessible={true}
        numberOfLines={0}
        style={
          Array [
            Object {
              "textAlign": "center",
            },
            Object {
              "color": "#000000",
            },
            Object {
              "fontSize": 21,
              "lineHeight": 31.5,
            },
            Object {
              "fontFamily": "BoschSans-Medium",
            },
            Object {},
          ]
        }
      >
        Temperature
      </Text>
    </View>
  </View>
  <Image
    source={
      Object {
        "testUri": "../../../src/assets/images/header_ribbon.png",
      }
    }
    style={
      Object {
        "height": 8,
        "width": "100%",
      }
    }
  />
  <RCTScrollView
    collapsable={false}
    contentContainerStyle={
      Object {
        "backgroundColor": "white",
        "flexGrow": 1,
      }
    }
    handlerTag={2}
    handlerType="NativeViewGestureHandler"
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
    waitFor={
      Array [
        Object {
          "current": null,
        },
      ]
    }
  >
    <View>
      <View
        style={
          Object {
            "flex": 1,
            "marginBottom": 20,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "marginTop": 24,
            }
          }
        >
          <svg
            data-file-name="SvgTemperatureSettings"
            fill="#000"
          />
        </View>
        <Text
          accessibilityLabel="Please set the temperature settings
for the thermostat"
          accessible={true}
          numberOfLines={0}
          style={
            Array [
              Object {
                "textAlign": "center",
              },
              Object {
                "color": "#000000",
              },
              Object {
                "fontSize": 16,
                "lineHeight": 24,
              },
              Object {
                "fontFamily": "BoschSans-Regular",
              },
              Object {
                "marginBottom": 22,
                "marginTop": 16,
              },
            ]
          }
        >
          Please set the temperature settings
for the thermostat
        </Text>
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "paddingHorizontal": 56,
              "paddingVertical": 10,
            }
          }
        >
          <View
            style={
              Object {
                "flex": 1,
                "height": 65,
              }
            }
          >
            <View
              style={
                Array [
                  undefined,
                  Object {
                    "flex": 1,
                    "flexDirection": "row",
                    "justifyContent": "center",
                    "marginVertical": 5,
                  },
                ]
              }
            >
              <View
                accessibilityHint="Activate to set the thermostat on fahrenheit mode."
                accessibilityLabel="°F option"
                accessibilityRole="button"
                accessible={true}
                collapsable={false}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "backgroundColor": "#FFFFFF",
                    "borderColor": "#004975",
                    "borderWidth": 1,
                    "flex": 1,
                    "opacity": 1,
                    "paddingVertical": 15,
                  }
                }
              >
                <Text
                  accessible={true}
                  numberOfLines={0}
                  style={
                    Array [
                      Object {
                        "textAlign": "center",
                      },
                      Object {
                        "color": "#004975",
                      },
                      Object {
                        "fontSize": 16,
                        "lineHeight": 24,
                      },
                      Object {
                        "fontFamily": "BoschSans-Regular",
                      },
                      Object {},
                    ]
                  }
                >
                  °F
                </Text>
              </View>
              <View
                accessibilityHint="Activate to set the thermostat on celcius mode."
                accessibilityLabel="°C option"
                accessibilityRole="button"
                accessible={true}
                collapsable={false}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "backgroundColor": "#004975",
                    "borderColor": "#004975",
                    "borderWidth": 1,
                    "flex": 1,
                    "opacity": 1,
                    "paddingVertical": 15,
                  }
                }
              >
                <Text
                  accessible={true}
                  numberOfLines={0}
                  style={
                    Array [
                      Object {
                        "textAlign": "center",
                      },
                      Object {
                        "color": "#FFFFFF",
                      },
                      Object {
                        "fontSize": 16,
                        "lineHeight": 24,
                      },
                      Object {
                        "fontFamily": "BoschSans-Regular",
                      },
                      Object {},
                    ]
                  }
                >
                  °C
                </Text>
              </View>
            </View>
          </View>
        </View>
        <View
          accessibilityActions={
            Array [
              Object {
                "name": "activate",
              },
            ]
          }
          accessibilityHint="Activate to turn system on automatically when the room temperature reaches above the set temperature points."
          accessibilityLabel="Auto-On"
          accessible={true}
          onAccessibilityAction={[Function]}
          style={
            Object {
              "borderBottomColor": "#BFC0C2",
              "borderBottomWidth": 1,
              "marginHorizontal": 16,
              "paddingBottom": 19,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "space-between",
                "marginBottom": 18,
              }
            }
          >
            <Text
              accessible={true}
              numberOfLines={0}
              style={
                Array [
                  Object {
                    "textAlign": "center",
                  },
                  Object {
                    "color": "#000000",
                  },
                  Object {
                    "fontSize": 16,
                    "lineHeight": 24,
                  },
                  Object {
                    "fontFamily": "BoschSans-Medium",
                  },
                  Object {},
                ]
              }
            >
              Auto-On
            </Text>
            <View
              style={
                Object {
                  "marginTop": -16,
                }
              }
            >
              <View
                accessible={true}
                collapsable={false}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
                testID="autoOption"
              >
                <View
                  collapsable={false}
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "rgba(113, 118, 124, 1)",
                      "borderRadius": 25,
                      "flexDirection": "row",
                      "height": 24,
                      "marginHorizontal": 1,
                      "marginTop": 16,
                      "padding": 5,
                      "width": 48,
                    }
                  }
                >
                  <View
                    collapsable={false}
                    style={Object {}}
                  >
                    <Text />
                  </View>
                  <View
                    collapsable={false}
                    style={
                      Object {
                        "backgroundColor": "rgba(255, 255, 255, 1)",
                        "borderRadius": 20,
                        "height": 12,
                        "transform": Array [
                          Object {
                            "translateX": 0,
                          },
                        ],
                        "width": 12,
                      }
                    }
                  >
                    <View
                      collapsable={false}
                      style={Object {}}
                    >
                      <Text />
                    </View>
                  </View>
                  <View
                    collapsable={false}
                    style={Object {}}
                  >
                    <Text />
                  </View>
                  <View
                    style={
                      Object {
                        "position": "absolute",
                        "right": 5,
                      }
                    }
                  />
                </View>
              </View>
            </View>
          </View>
          <Text
            accessible={true}
            numberOfLines={0}
            style={
              Array [
                Object {
                  "textAlign": "left",
                },
                Object {
                  "color": "#000000",
                },
                Object {
                  "fontSize": 16,
                  "lineHeight": 24,
                },
                Object {
                  "fontFamily": "BoschSans-Regular",
                },
                Object {},
              ]
            }
          >
            Turn system on automatically when the
room temperature reaches above the set
temperature points.
          </Text>
        </View>
        <View
          accessibilityActions={
            Array [
              Object {
                "name": "activate",
              },
            ]
          }
          accessibilityHint="Activate to send an alert message when room temperature is out of the set temperature point."
          accessibilityLabel="Alert Message"
          accessible={true}
          onAccessibilityAction={[Function]}
          style={
            Object {
              "marginHorizontal": 16,
              "marginTop": 16,
              "paddingBottom": 19,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "space-between",
                "marginBottom": 18,
              }
            }
          >
            <Text
              accessible={true}
              numberOfLines={0}
              style={
                Array [
                  Object {
                    "textAlign": "center",
                  },
                  Object {
                    "color": "#000000",
                  },
                  Object {
                    "fontSize": 16,
                    "lineHeight": 24,
                  },
                  Object {
                    "fontFamily": "BoschSans-Medium",
                  },
                  Object {},
                ]
              }
            >
              Alert Message
            </Text>
            <View
              style={
                Object {
                  "marginTop": -16,
                }
              }
            >
              <View
                accessible={true}
                collapsable={false}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
                testID="alertOption"
              >
                <View
                  collapsable={false}
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "rgba(113, 118, 124, 1)",
                      "borderRadius": 25,
                      "flexDirection": "row",
                      "height": 24,
                      "marginHorizontal": 1,
                      "marginTop": 16,
                      "padding": 5,
                      "width": 48,
                    }
                  }
                >
                  <View
                    collapsable={false}
                    style={Object {}}
                  >
                    <Text />
                  </View>
                  <View
                    collapsable={false}
                    style={
                      Object {
                        "backgroundColor": "rgba(255, 255, 255, 1)",
                        "borderRadius": 20,
                        "height": 12,
                        "transform": Array [
                          Object {
                            "translateX": 0,
                          },
                        ],
                        "width": 12,
                      }
                    }
                  >
                    <View
                      collapsable={false}
                      style={Object {}}
                    >
                      <Text />
                    </View>
                  </View>
                  <View
                    collapsable={false}
                    style={Object {}}
                  >
                    <Text />
                  </View>
                  <View
                    style={
                      Object {
                        "position": "absolute",
                        "right": 5,
                      }
                    }
                  />
                </View>
              </View>
            </View>
          </View>
          <Text
            accessible={true}
            numberOfLines={0}
            style={
              Array [
                Object {
                  "textAlign": "left",
                },
                Object {
                  "color": "#000000",
                },
                Object {
                  "fontSize": 16,
                  "lineHeight": 24,
                },
                Object {
                  "fontFamily": "BoschSans-Regular",
                },
                Object {},
              ]
            }
          >
            Send an alert message when room
temperature is out of the set temperature
point.
          </Text>
        </View>
        <View
          style={
            Object {
              "marginHorizontal": 16,
            }
          }
        >
          <Text
            accessible={true}
            numberOfLines={0}
            style={
              Array [
                Object {
                  "textAlign": "left",
                },
                Object {
                  "color": "#000000",
                },
                Object {
                  "fontSize": 16,
                  "lineHeight": 24,
                },
                Object {
                  "fontFamily": "BoschSans-Medium",
                },
                Object {
                  "marginBottom": 18,
                },
              ]
            }
          >
            Temperature
          </Text>
          <View
            style={
              Object {
                "marginBottom": 16,
              }
            }
          >
            <View
              style={
                Object {
                  "width": "100%",
                }
              }
            >
              <View
                accessibilityHint="Activate to open a modal and select the minimum temperature to send an alert."
                accessible={true}
                collapsable={false}
                focusable={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "borderBottomWidth": 1,
                      "height": 56,
                      "paddingLeft": 10,
                      "paddingRight": 16,
                      "width": "100%",
                    },
                    Object {
                      "backgroundColor": "#EFF1F2",
                    },
                    Object {
                      "flexDirection": "row",
                      "justifyContent": "space-between",
                    },
                  ]
                }
                testID="lowTemp"
              >
                <View>
                  <Text
                    accessible={true}
                    numberOfLines={0}
                    style={
                      Array [
                        Object {
                          "textAlign": "left",
                        },
                        Object {
                          "color": "#000000",
                        },
                        Object {
                          "fontSize": 12,
                          "lineHeight": 18,
                        },
                        Object {
                          "fontFamily": "BoschSans-Regular",
                        },
                        Object {},
                      ]
                    }
                  >
                    Low
                  </Text>
                  <Text
                    accessible={true}
                    numberOfLines={0}
                    style={
                      Array [
                        Object {
                          "textAlign": "left",
                        },
                        Object {
                          "color": "#000000",
                        },
                        Object {
                          "fontSize": 16,
                          "lineHeight": 24,
                        },
                        Object {
                          "fontFamily": "BoschSans-Regular",
                        },
                        Object {},
                      ]
                    }
                  >
                    undefined°C
                  </Text>
                </View>
                <svg
                  data-file-name="SvgTemperature1"
                  fill="#000"
                />
              </View>
              <Modal
                animationType="slide"
                hardwareAccelerated={false}
                onRequestClose={[Function]}
                transparent={false}
                visible={false}
              />
            </View>
          </View>
          <View
            style={
              Object {
                "width": "100%",
              }
            }
          >
            <View
              accessibilityHint="Activate to open a modal and select the maximum temperature to send an alert."
              accessible={true}
              collapsable={false}
              focusable={true}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "borderBottomWidth": 1,
                    "height": 56,
                    "paddingLeft": 10,
                    "paddingRight": 16,
                    "width": "100%",
                  },
                  Object {
                    "backgroundColor": "#EFF1F2",
                  },
                  Object {
                    "flexDirection": "row",
                    "justifyContent": "space-between",
                  },
                ]
              }
              testID="highTemp"
            >
              <View>
                <Text
                  accessible={true}
                  numberOfLines={0}
                  style={
                    Array [
                      Object {
                        "textAlign": "left",
                      },
                      Object {
                        "color": "#000000",
                      },
                      Object {
                        "fontSize": 12,
                        "lineHeight": 18,
                      },
                      Object {
                        "fontFamily": "BoschSans-Regular",
                      },
                      Object {},
                    ]
                  }
                >
                  High
                </Text>
                <Text
                  accessible={true}
                  numberOfLines={0}
                  style={
                    Array [
                      Object {
                        "textAlign": "left",
                      },
                      Object {
                        "color": "#000000",
                      },
                      Object {
                        "fontSize": 16,
                        "lineHeight": 24,
                      },
                      Object {
                        "fontFamily": "BoschSans-Regular",
                      },
                      Object {},
                    ]
                  }
                >
                  undefined°C
                </Text>
              </View>
              <svg
                data-file-name="SvgTemperature1"
                fill="#000"
              />
            </View>
            <Modal
              animationType="slide"
              hardwareAccelerated={false}
              onRequestClose={[Function]}
              transparent={false}
              visible={false}
            />
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
